tensorflow/contrib/learn/python/learn/learn_io/data_feeder.py

```diff
@@ -459,19 +545,38 @@ def __init__(self, x, y, n_classes, batch_size):
       y_first_el = None
       self._y = None
     self.n_classes = n_classes
-    x_first_el = ops.convert_to_tensor(x_first_el)
-    y_first_el = ops.convert_to_tensor(y_first_el) if y is not None else None
-    self.input_shape, self.output_shape, self._batch_size = _get_in_out_shape(
-        [1] + list(x_first_el.get_shape()),
-        [1] + list(y_first_el.get_shape()) if y is not None else None,
-        n_classes,
-        batch_size)
-    self._input_dtype = _check_dtype(x_first_el.dtype).as_numpy_dtype
+
+    x_is_dict, y_is_dict = isinstance(x_first_el, dict), y is not None and isinstance(y_first_el, dict)
+    if y_is_dict and n_classes is not None:
+      assert(isinstance(n_classes, dict))
```

Comment:
I have updated the DocString so it makes more sense, but the idea is that if `y` is dictionary, then `n_classes` can be either `None` ( in which case none of the labels in `y` are converted to one-hot) or must be dictionary with same keys as those in `y`.  Hope it makes more sense now?